crypto/key_store_passphrase.go

```diff
@@ -112,40 +113,44 @@ func (ks keyStorePassphrase) GetKey(keyAddr []byte, auth string) (key *Key, err
 	return key, err
 }
 
-func (ks keyStorePassphrase) GetKeyAddresses() (addresses [][]byte, err error) {
+func (ks keyStorePassphrase) GetKeyAddresses() (addresses []common.Address, err error) {
 	return GetKeyAddresses(ks.keysDirPath)
 }
 
 func (ks keyStorePassphrase) StoreKey(key *Key, auth string) (err error) {
 	authArray := []byte(auth)
-	salt := randentropy.GetEntropyMixed(32)
+	salt := randentropy.GetEntropyCSPRNG(32)
```

Comment:
@subtly The length is indirectly checked here: https://github.com/Gustav-Simonsson/go-ethereum/blob/improve_key_store_crypto/crypto/randentropy/rand_entropy.go#L30 as io.ReadFull returns an error if fewer than requested bytes are read: http://golang.org/pkg/io/#ReadFull